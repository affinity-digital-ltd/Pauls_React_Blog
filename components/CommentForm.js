import { Component } from 'react'

class CommentForm extends Component {
  render () {
    const { userProfile } = this.props

    return <div className='c-comments-form'>
      <form onSubmit={this.props.postComment}>
        <input type='hidden' name='authorName' value={userProfile.authorName} />
        <input type='text' name='body' className='c-comments-form__body-text' placeholder='Add your comment here' required />
        <input type='hidden' name='image' value={userProfile.image} />
        <input type='submit' value='Post' className='c-comments-form__submit-btn' />
      </form>
    </div>
  }
}

export default CommentForm
