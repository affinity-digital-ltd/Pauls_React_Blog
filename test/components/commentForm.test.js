import CommentForm from '../../components/CommentForm'

describe('<CommentForm />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <CommentForm userProfile={'Paul'} />
    )

    expect(component).toMatchSnapshot()
  })
})
