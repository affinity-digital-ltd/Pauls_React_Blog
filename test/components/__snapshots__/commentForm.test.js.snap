// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentForm
    userProfile="Paul"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={undefined}
      >
        <input
          name="authorName"
          type="hidden"
          value={undefined}
        />
        <input
          className="c-comments-form__body-text"
          name="body"
          placeholder="Add your comment here"
          required={true}
          type="text"
        />
        <input
          name="image"
          type="hidden"
          value={undefined}
        />
        <input
          className="c-comments-form__submit-btn"
          type="submit"
          value="Post"
        />
      </form>,
      "className": "c-comments-form",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="authorName"
            type="hidden"
            value={undefined}
          />,
          <input
            className="c-comments-form__body-text"
            name="body"
            placeholder="Add your comment here"
            required={true}
            type="text"
          />,
          <input
            name="image"
            type="hidden"
            value={undefined}
          />,
          <input
            className="c-comments-form__submit-btn"
            type="submit"
            value="Post"
          />,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "authorName",
            "type": "hidden",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-comments-form__body-text",
            "name": "body",
            "placeholder": "Add your comment here",
            "required": true,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "image",
            "type": "hidden",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-comments-form__submit-btn",
            "type": "submit",
            "value": "Post",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={undefined}
        >
          <input
            name="authorName"
            type="hidden"
            value={undefined}
          />
          <input
            className="c-comments-form__body-text"
            name="body"
            placeholder="Add your comment here"
            required={true}
            type="text"
          />
          <input
            name="image"
            type="hidden"
            value={undefined}
          />
          <input
            className="c-comments-form__submit-btn"
            type="submit"
            value="Post"
          />
        </form>,
        "className": "c-comments-form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="authorName"
              type="hidden"
              value={undefined}
            />,
            <input
              className="c-comments-form__body-text"
              name="body"
              placeholder="Add your comment here"
              required={true}
              type="text"
            />,
            <input
              name="image"
              type="hidden"
              value={undefined}
            />,
            <input
              className="c-comments-form__submit-btn"
              type="submit"
              value="Post"
            />,
          ],
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "authorName",
              "type": "hidden",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-comments-form__body-text",
              "name": "body",
              "placeholder": "Add your comment here",
              "required": true,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "image",
              "type": "hidden",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-comments-form__submit-btn",
              "type": "submit",
              "value": "Post",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
